### ISSUE TITLE
Map.invalidateSize() creates incorrect zoom in Safari 9.0.1's "Responsive design" mode

### ISSUE THREAD

**Issue by aurasia** [Leaflet/Leaflet#4081]
Running Leaflet 0.7.7 in Safari 9.0.1 on a Mac

When a Leaflet map is displayed in "responsive design" mode (under Safari's Developer menu), a call to `Map.invalidateSize()` may zoom out way too much, rendering this mode unusable for cross-device testing.
Based on some initial debugging, I believe the culprit is the call to `this._container.clientWidth` (and `this._container.clientHeight`) inside `Map.getSize()`. These attributes return values that are way too high under certain conditions.

Using the Chrome equivalent developer tool (called "Emulation") does not seem to have this problem.

**Comment by mourner** [comment]
Do you have a JSFiddle test case for this or is this reproducible on any Leaflet map, e.g. on leafletjs.com? Generally this looks like a Safari bug rather than something that can be fixed on the Leaflet side.

**Comment by aurasia** [comment]
Ok, I tried for a long time in JSFiddle, but unfortunately, I could not get a minimum sample to reproduce the bug.

Will close it for now and open a new issue if/when I am able to reproduce. Thanks

**Comment by aurasia** [comment]
Ok, I think I know what happened - I had some visible DOM elements outside the viewport, and while Chrome "Emulation" just does not display them at all, Safari "Responsive Design" tries to show them somehow by some zooming (but not enough zooming, so I did not understand what was happening).

When displayed on an actual iPhone device, both Chrome and Safari just zoom out completely to display all DOM elements, so I finally understood what was happening. I now set `display: none;` for these DOM elements and everything is fine.

Lesson of the day: Chrome "Emulation" and Safari "Responsive Design" are nice-to-have, but do NOT exactly match browsers on actual devices. When in doubt, check an actual device.

Sorry for wasting your time. And a big THANK YOU for Leaflet, really love it!
